src/NJsonSchema.CodeGeneration.CSharp/Models/PropertyModel.cs

```diff
@@ -33,6 +33,34 @@ public class PropertyModel : PropertyModelBase
             _property = property;
             _settings = settings;
             _resolver = typeResolver;
+
+            if (_property.ActualSchema.Minimum.HasValue || _property.ActualSchema.Maximum.HasValue)
+            {
+                if (string.IsNullOrEmpty(_property.ActualSchema.Format))
+                {
+                    if (_property.ActualSchema.Type == JsonObjectType.Integer)
+                    {
+                        // If an integer with no format is specified and then test Minimum and Maximum values are compatible with int32 and if not then explicitly set format to int64
+                        if (IsOutOfRange<int>(_property.ActualSchema.Minimum.GetValueOrDefault()) || IsOutOfRange<int>(_property.ActualSchema.Maximum.GetValueOrDefault()))
+                        {
+                            _property.ActualSchema.Format = JsonFormatStrings.Long;
```

Comment:
Just merged but now i see that you are modifying the schema in the generator! This is a big no-go as a generator should not have side-effects. Now i need to find a way to fix this...